src/python/gudhi/representations/archipelago.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Martin Royer
+
+import copy
+
+import numpy as np
+import pandas as pd
+
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from gudhi.representations.vector_methods import Atol, TopologicalVector, BettiCurve
+
+
+class Archipelago(BaseEstimator, TransformerMixin):
+    """
+    Transformer that dictionary-wraps persistence diagram vectorizers, i.e. objects from gudhi.representations.vector_methods.
+    One provides persistence diagram vectorizers (by way of either `island` or `island_dict`), and the Archipelago object will
+    |fit on| and |transform = vectorize| list or series of persistence diagrams (in pandas format).
+    The object is sklearn-API consistent.
+
+    Parameters:
+        island: island for populating archipelago, i.e. object to vectorize the target in each homology
+            dimensions. Must be `copy.deepcopy`-able. *Will be ignored if island_list is given*.
+        island_dict: island dict for populating archipelago, i.e. dictionary of objects to vectorize persistence
+            diagrams according to homology_dimensions.
+
+    Examples
+    >>> pdiagram1 = [(0, (0.0, 2.34)), (0, (0.0, 0.956)), (1, (0.536, 0.856)), (2, (1.202, 1.734))]
```

Comment:
I am a bit reluctant to add more code that uses the old format for persistence diagrams, when we are trying to replace it with something more convenient (see #925, https://github.com/GUDHI/gudhi-devel/blob/master/src/python/gudhi/sklearn/cubical_persistence.py or #691). Of course we could change archipelago later to accept the new format, but it may become complicated if we try to be compatible with both.

Reply:
I'm not sure what the new format is but it seems it will be better optimized for Archipelago, so all the better. Do you mean you want me to wait for those PR to come online before processing this one?

Reply:
> I'm not sure what the new format is but it seems it will be better optimized for Archipelago, so all the better.

It is the one we discussed before the holidays. Instead of a list of (dim,(birth,death)), if the user asked for homology in dimensions [i,j,k], they get [dgm_i, dgm_j, dgm_k] where each dgm is a numpy array of shape (*,2).

It is better in that you already have numpy arrays for each dimension. However, because it is a list and not a dict, you do not have access to the labels [i,j,k], the user has to provide them separately if you need them for something. It also makes it ambiguous, if we refer to the 0-th diagram, whether it corresponds to homology in dimension 0, or in dimension i (the first of the list).

> Do you mean you want me to wait for those PR to come online before processing this one?

I don't know what I want :wink: